### PR TITLE
[uikit] Update for Xcode 9 beta 6 - Part 4

### PR DESCRIPTION
**Pull request by VincentDondain**
The following headers are now up to date:

- UICollectionViewFlowLayout.h
- UIDocumentPickerViewController.h
- UIFontDescriptor.h
- UIImage.h
- UIImageView.h
- UIImagePickerController.h

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -6923,12 +6933,48 @@ interface UIActivityIndicatorView : NSCoding {
...
+		CGSize PreferredPresentationSizeForItemProvider { get; }
+	}
+
+	[iOS (11,0), TV (11,0), Watch (4,0)]
+	[Category]
+	[BaseType (typeof(NSItemProvider))]
+	interface NSItemProvider_UIKitAdditions {
+		[NoWatch, NoTV]
+		[NullAllowed, Export ("teamData", ArgumentSemantic.Copy)]
+		NSData GetTeamData ();
```
Comment (rolfbjarne) [inline src/uikit.cs]: Maybe this should be inlined in `NSItemProvider` instead, that would make it possible to use actual properties instead of Get/Set methods. @spouliot, what do you think?

Comment (spouliot) [inline src/uikit.cs]: Good idea, we already have `#if MONOMAC` block for `NSItemProvider` to do the same for XM/macOS.

Comment (spouliot) [inline src/uikit.cs]: @VincentDondain let's do that in a separate PR so the other bindings can ship in this week's preview.

**Comment by spouliot** [comment]
known tvOS failure with FontDescriptorTest.WithFeature -> https://bugzilla.xamarin.com/show_bug.cgi?id=58929

File: src/uikit.cs (hunk 1 above)
Comment (VincentDondain) [inline src/uikit.cs]: Fixed in https://github.com/xamarin/xamarin-macios/pull/2538
